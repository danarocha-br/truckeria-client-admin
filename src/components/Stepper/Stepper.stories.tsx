import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import { Stepper, StepperProps } from '.';
import { Box } from '../Box';

const steps = [
  {
    label: 'Your info',
    // isCompleted: true,
  },
  {
    label: 'Food truck info',
    // isActive: true,
  },
  {
    label: 'Contact info',
  },
];

export default {
  title: 'Navigation/Stepper',
  component: Stepper,
  parameters: {
    layout: 'centered',
    status: 'stable', // beta| deprecated | review | stable,
    design: {
      type: 'figma',
      url: '',
    },
  },
  args: {
    steps: steps,
    activeIndex: 0,
  },
} as Meta;

const Template: Story<StepperProps> = (args) => (
  <Box css={{ w: 500 }}>
    <Stepper {...args} />
  </Box>
);

export const Primary = Template.bind({});
