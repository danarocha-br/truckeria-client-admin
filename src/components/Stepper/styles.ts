import { styled } from '../../../stitches.config';

export const Container = styled('nav', {
  w: '$full',
});

export const List = styled('ol', {
  d: 'flex',
  alignItems: 'center',
  justify: 'space-between',
  w: '$full',
  p: 0,

  '&:hover li:not(:hover) span': {
    opacity: 0,
  },
});

export const Item = styled('li', {
  color: '$text-lighter',
  textAlign: 'center',
  position: 'relative',
  whiteSpace: 'nowrap',
  minWidth: 0,
  borderBottom: '3px solid $surface-base-default',
  pb: '$6',
  overflow: 'visible',
  listStyle: 'none',
  flex: 2,

  '&:before': {
    content: '',
    bg: '$surface-base-subdued',
    d: 'block',
    w: 18,
    h: 18,
    borderRadius: '$round',
    border: '3px solid',
    borderColor: '$background-default',
    position: 'absolute',
    bottom: -9,
    left: 'calc(50% - 6px)',
    zIndex: 3,
    transitionPoperty: 'all',
    transitionTimingDunction: 'cubic-bezier(0.4, 0, 0.2, 1)',
    transitionDuration: '150ms',
  },

  '&:first-child, &:last-child': {
    flex: 1,
  },

  '&:first-child': {
    textAlign: 'left',

    '&:before': {
      left: -1,
    },
  },

  '&:last-child': {
    textAlign: 'right',

    '&:before': {
      right: '$-1',
      left: 'auto',
    },

    '&:after': {
      w: 'inherit',
      left: 'inherit',
    },
  },

  '& span': {
    transition: 'opacity cubic-bezier(0.4, 0, 0.2, 1) 300ms',
    opacity: 0,
  },

  '&:not(:first-child):after': {},

  '&:hover': {
    '&:before': {
      transform: 'scale(1.2)',
    },
    '& span': {
      opacity: 0.5,
    },
  },

  variants: {
    isActive: {
      true: {
        '&:before': {
          bg: '$background-default',
          borderColor: '$action-success-default',
        },

        '&:last-child:after': {
          w: '200%',
          left: '-100%',
        },

        '& span': {
          opacity: 1,
        },

        '&:not(:first-child):after': {
          content: '',
          d: 'block',
          w: '$full',
          position: 'absolute',
          left: '-50%',
          bottom: '-2px',
          borderBottom: '3px solid $action-success-default',
        },

        '&:hover:before': {
          bg: '$background-default',
          borderColor: '$action-success-default',
        },
      },
    },
    isCompleted: {
      true: {
        '&:before': {
          bg: '$action-success-default',
        },

        '&:last-child:after': {
          w: '200%',
          left: '-100%',
        },

        '&:not(:first-child):after': {
          content: '',
          d: 'block',
          w: '$full',
          position: 'absolute',
          left: '-50%',
          bottom: '$-1',
          borderBottom: '2px solid $action-success-default',
        },
      },
    },
  },
});
