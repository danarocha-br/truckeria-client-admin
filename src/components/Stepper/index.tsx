import React from 'react';

import * as S from './styles';

export type StepperItemProps = {
  label: string;
  isActive?: boolean;
  isCompleted?: boolean;
};

export type StepperProps = {
  steps: StepperItemProps[];
  activeIndex?: number;
};

export const Stepper = ({ steps, activeIndex }: StepperProps) => {
  return (
    <S.Container>
      <S.List>
        {steps &&
          steps.map((item: StepperItemProps, index: number) => (
            <S.Item
              key={index}
              isActive={item.isActive || activeIndex === index}
              isCompleted={
                //@ts-ignore
                item.isCompleted || activeIndex > index
              }
              aria-label={item.label}
            >
              <span> {item.label}</span>
            </S.Item>
          ))}
      </S.List>
    </S.Container>
  );
};
