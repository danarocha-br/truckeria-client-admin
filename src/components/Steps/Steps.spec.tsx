import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import Stepper from '.';

const steps = [
  {
    label: 'Basic info',
    to: '/',
  },
  {
    label: 'Contact info',
    to: '/',
    isActive: true,
  },
  {
    label: 'Some other info',
    to: '/',
  },
];

describe('<Stepper />', () => {
  it('should render the Stepper', () => {
    const { container } = renderWithTheme(<Stepper steps={steps} />);

    expect(screen.getByLabelText(/Contact info/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
