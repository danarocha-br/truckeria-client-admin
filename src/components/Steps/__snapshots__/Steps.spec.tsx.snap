// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stepper /> should render the Stepper 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1:hover li:not(:hover) span {
  opacity: 0;
}

.c2 {
  text-align: center;
  position: relative;
  white-space: nowrap;
  min-width: 0px;
  border-bottom-width: 2px;
  padding-bottom: 0.75rem;
  overflow: visible;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  border-color: #ECEFF4;
}

.c2:first-child,
.c2:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2:first-child {
  text-align: left;
}

.c2:first-child:before {
  left: -0.25rem;
}

.c2:last-child {
  text-align: right;
}

.c2:last-child:before {
  right: -0.25rem;
  left: auto;
}

.c2:last-child:after {
  width: inherit;
  left: inherit;
}

.c2:before {
  display: block;
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  border-width: 2px;
  position: absolute;
  bottom: -0.5rem;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  content: '';
  left: calc(50% - 6px);
  z-index: 3;
  background-color: #B8C4C8;
  border-color: #FFFFFF;
}

.c2 span {
  -webkit-transition-property: opacity;
  transition-property: opacity;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  opacity: 0;
}

.c2:not(:first-child):after {
  bottom: -2px;
  content: '';
}

.c2:hover:before {
  -webkit-transform: scale(1.33);
  -ms-transform: scale(1.33);
  transform: scale(1.33);
}

.c2:hover span {
  opacity: 1;
}

.c3 {
  text-align: center;
  position: relative;
  white-space: nowrap;
  min-width: 0px;
  border-bottom-width: 2px;
  padding-bottom: 0.75rem;
  overflow: visible;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  border-color: #ECEFF4;
}

.c3:first-child,
.c3:last-child {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c3:first-child {
  text-align: left;
}

.c3:first-child:before {
  left: -0.25rem;
}

.c3:last-child {
  text-align: right;
}

.c3:last-child:before {
  right: -0.25rem;
  left: auto;
}

.c3:last-child:after {
  width: 200%;
  left: -100%;
}

.c3:before {
  display: block;
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  border-width: 2px;
  position: absolute;
  bottom: -0.5rem;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  content: '';
  left: calc(50% - 6px);
  z-index: 3;
  background-color: #FFFFFF;
  border-color: #76C248;
}

.c3 span {
  -webkit-transition-property: opacity;
  transition-property: opacity;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  opacity: 1;
}

.c3:not(:first-child):after {
  display: block;
  width: 100%;
  position: absolute;
  left: -50%;
  border-bottom-width: 2px;
  border-style: solid;
  --tw-border-opacity: 1;
  border-color: rgba(118,194,72,var(--tw-border-opacity));
  bottom: -2px;
  content: '';
}

.c3:hover:before {
  background-color: #FFFFFF;
  border-color: #76C248;
  -webkit-transform: scale(1.33);
  -ms-transform: scale(1.33);
  transform: scale(1.33);
}

.c3:hover span {
  opacity: 1;
}

<nav
  class="c0"
>
  <ol
    class="c1"
  >
    <li
      aria-label="Basic info"
      class="c2"
    >
      <span>
         
        Basic info
      </span>
    </li>
    <li
      aria-label="Contact info"
      class="c3"
    >
      <span>
         
        Contact info
      </span>
    </li>
    <li
      aria-label="Some other info"
      class="c2"
    >
      <span>
         
        Some other info
      </span>
    </li>
  </ol>
</nav>
`;
